### ISSUE TITLE
Exception thrown in the Developer Tools Console when editing a non-folder-scoped Dart file

### ISSUE THREAD

**Issue by Porl91** [Dart-Code/Dart-Code#599]
My versions:
DartCode: 2.9.0-beta.3
Dart: 1.24.3
VS Code: 1.20.1

I noticed that there's an exception thrown in the Developer Tools Console when saving a Dart file (outside of a folder), then entering a character into the document.

```
ERR Cannot read property 'uri' of undefined: TypeError: Cannot read property 'uri' of undefined
	at Object.isFlutterProject (C:\Users\Porl\.vscode\extensions\Dart-Code.dart-code-2.9.0-beta.3\out\src\utils.js:22:39)
	at SnippetCompletionItemProvider.context.subscriptions.push.vs.languages.registerCompletionItemProvider.snippet_completion_item_provider_1.SnippetCompletionItemProvider [as shouldRender] (C:\Users\Porl\.vscode\extensions\Dart-Code.dart-code-2.9.0-beta.3\out\src\extension.js:164:198)
	at SnippetCompletionItemProvider.provideCompletionItems (C:\Users\Porl\.vscode\extensions\Dart-Code.dart-code-2.9.0-beta.3\out\src\providers\snippet_completion_item_provider.js:27:18)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:633:20
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:98:622
	at new n.Class.derive._oncancel (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:75:657)
	at Object.t.asWinJsPromise (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:98:585)
	at e.provideCompletionItems (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:632:944)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:643:654
	at e._withAdapter (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:637:537)
```

It's probably the same issue, but when I press F5 it generates a similar error:

```
Cannot read property 'uri' of undefined: TypeError: Cannot read property 'uri' of undefined
	at Object.isFlutterProject (C:\Users\Porl\.vscode\extensions\Dart-Code.dart-code-2.9.0-beta.3\out\src\utils.js:22:39)
	at DebugConfigProvider.resolveDebugConfiguration (C:\Users\Porl\.vscode\extensions\Dart-Code.dart-code-2.9.0-beta.3\out\src\providers\debug_config_provider.js:24:35)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:536:10
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:98:622
	at new n.Class.derive._oncancel (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:75:657)
	at Object.t.asWinJsPromise (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:98:585)
	at e.$resolveDebugConfiguration (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:535:947)
	at e._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:584:983)
	at e._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:584:701)
	at e._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:583:836)
```

I've tested that both of these issues don't occur when the Dart file is nested inside a folder. 😀 

I noticed this because in the Dart file I was editing I also wasn't receiving any warnings back from the analyser. Could this be related?

Here's a PS of the project structure + open document
![wat1](https://user-images.githubusercontent.com/13272784/36442804-cf3686e6-166e-11e8-8c38-c315a45de8a6.png)
 if that helps.

**Comment by DanTup** [comment]
I'll fix the error in v2.9, but depending on how much work it is to support files without a folder open, that might have to wait till next version.

**Comment by DanTup** [comment]
I've split support for Dart files outside of folders into #602 because it needs an analyzer change. This means you won't get completion etc.; but I'll at least fix the errors for this.
